lib/screens_web/controllers/v2/screen_api_controller.ex

```diff
@@ -80,7 +80,14 @@ defmodule ScreensWeb.V2.ScreenApiController do
           screen_side
         )
 
-        json(conn, ScreenData.by_screen_id(screen_id))
+        json(
+          conn,
+          ScreenData.by_screen_id(screen_id,
+            is_real_screen: is_screen,
+            screen_id: screen_id,
+            triptych_pane: triptych_pane
```

Comment:
I think one of the advantages to using an `opts` list is that they can be arbitrary in length and a function doing a lookup in `opts` only needs to look for what they know they need. Similar to `params` in this action. This action is used by all screen types but references very specific values only needed by certain screen types. Because a lookup of key that doesn't exist is safe for these lists, we just need to do some `nil` checks before using them. 

I definitely see where you are coming from though. My biggest hesitation with conditionally adding it is that it will make the code here a little more complex for no drastic benefit. What do you think?

Reply:
Ok, I can see that. Works for me!